pkg/flow/internal/controller/component.go

```diff
@@ -137,21 +141,6 @@ func NewComponentNode(globals ComponentGlobals, b *hcl.Block) *ComponentNode {
 	return cn
 }
 
-func getRegistration(id ComponentID) (component.Registration, bool) {
-	// id is the fully qualified name of the component, including the custom user
-	// identifier, if supported by the component. We don't know if the component
-	// we're looking up is a singleton or not, so we have to try the lookup
-	// twice: once with all name fragments in the ID and once without the last
-	// one (e.g., the one that represents the user ID).
-	reg, ok := component.Get(id.String())
-	if ok {
-		return reg, ok
-	}
-
-	reg, ok = component.Get(id[:len(id)-1].String())
-	return reg, ok
-}
```

Comment:
This is no longer necessary because now we can cleanly tell what part of a block is a user label and what part is the name, unlike before where the label slice contained part of a user label and part of the component name.